packages/calcite-components/src/components/list/list.tsx

```diff
@@ -160,7 +171,7 @@ export class List
 
   @state() assistiveText: string;
 
-  @state() dataForFilter: ItemData = [];
+  @state() dataForFilter: (ItemData | ItemGroupData)[] = [];
```

Comment:
Instead of including separate group data, can we just add the group data to each item within a group? It would just be the `heading` property added to each item.

If we just added group heading to the itemData, then we wouldn't need the logic in `filterGroupItems`

Theres already existing logic to show a group if one or more items match and hide a group if not. The only part missing is the group data attached to each item.

Reply:
The above wouldn't work where only group heading is matched (nested group) but none of the items or ancestor group have a matching text.

Reply:
@anveshmekala it should work.

> The current filtering only applies to items, but should also include groups and any matches should show all items under a matching group.

**any matches should show all items under a matching group**

According to the request, if a group matches, all items under that group should also match.

Reply:
Its working as expected! 🎉